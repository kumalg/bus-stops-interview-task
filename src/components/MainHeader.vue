<template>
    <header class="main-header">
        <h1 class="main-header-title">{{ title }}</h1>
    </header>
</template>

<script setup lang="ts">
defineProps<{
    title: string;
}>();
</script>

<style lang="scss" scoped>
.main-header {
    padding: 0.25rem 0;

    .main-header-title {
        font-weight: 600;
        font-size: 1.5rem;
        line-height: 2rem;
        margin: 0;
    }
}
</style>
