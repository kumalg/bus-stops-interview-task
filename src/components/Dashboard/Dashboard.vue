<template>
    <div class="dashboard">
        <MainHeader title="Timetable" class="main-header" />

        <NavigationBar />

        <main class="main-content">
            <RouterView v-slot="{ Component }">
                <Transition name="fade" mode="out-in">
                    <component :is="Component" />
                </Transition>
            </RouterView>
        </main>
    </div>
</template>

<script setup lang="ts">
import { RouterView } from 'vue-router';

import MainHeader from '@/components//MainHeader/MainHeader.vue';
import NavigationBar from '@/components/NavigationBar/NavigationBar.vue';
</script>

<style lang="scss" scoped>
.dashboard {
    max-width: 1376px;
    margin: 0 auto;
    height: 100%;
    display: flex;
    flex-direction: column;
    gap: 1rem;

    .main-header {
        margin-bottom: 0.5rem;
    }

    .main-content {
        flex: 1;
        min-height: 0;
    }
}
</style>
