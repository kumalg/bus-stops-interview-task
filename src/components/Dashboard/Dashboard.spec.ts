import { shallowMount } from '@vue/test-utils';

import Dashboard from '@/components/Dashboard/Dashboard.vue';
import MainHeader from '@/components/MainHeader/MainHeader.vue';

describe('Dashboard', () => {
    const DashboardFactory = () => {
        return shallowMount(Dashboard);
    };

    it('renders proper html markup', () => {
        const wrapper = DashboardFactory();

        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders Timetable title', () => {
        const wrapper = DashboardFactory();

        expect(wrapper.findComponent(MainHeader).props()['title']).toBe('Timetable');
    });
});
