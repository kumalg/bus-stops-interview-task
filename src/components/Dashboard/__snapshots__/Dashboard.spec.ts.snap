// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders proper html markup 1`] = `
<div
  class="dashboard"
>
  <main-header-stub
    class="main-header"
    title="Timetable"
  />
  <navigation-bar-stub />
  <main
    class="main-content"
  >
    <router-view-stub
      name="default"
    />
  </main>
</div>
`;
